conf/application.conf

```diff
@@ -17,21 +17,40 @@ play {
     """
   }
-  filters.headers {
-    # Production Content Security Policy
-    # The sha256 hash in the script-src refers to the google analytics configuration script in main.scala.html.
-    # contentSecurityPolicy = "default-src 'self'; script-src 'self' 'unsafe-eval' *.googletagmanager.com app.olvy.co webknossos.olvy.co *.gstatic.com 'sha256-NA81o2FV8jh0UizSPsBz2qEGpXfap54Eghuo7vxrH8g='; style-src 'self' 'unsafe-inline' *.gstatic.com; connect-src 'self' app.olvy.co api.github.com gist.github.com *.google-analytics.com api.airbrake.io; frame-src webknossos.olvy.co; img-src 'self' data:"
-    # Development Content Security Policy
-    # In addition to the production policies, the development policy contains the script-src 'unsafe-inline' (to make the React Dev Tools work in Firefox) and the connect-src 'ws://localhost:9002' (webpack dev-server).
-    contentSecurityPolicy = "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' *.googletagmanager.com  app.olvy.co webknossos.olvy.co *.gstatic.com; style-src 'self' 'unsafe-inline' *.gstatic.com; connect-src 'self' ws://localhost:9002 app.olvy.co api.github.com gist.github.com *.google-analytics.com api.airbrake.io; frame-src webknossos.olvy.co; img-src 'self' data:"
-    contentTypeOptions = null
-    frameOptions = null
-    xssProtection = null # Not supported on modern browsers
+  filters {
+    # Note that the CSP filter is not enabled. We manually add this header ourselves for the
+    # main view, using the following config (note that currently only directives are used, not nonce or hashes)
+    csp.directives {
+      default-src = "'self'"
+      # In production 'unsafe-inline' should be removed as it is only needed to make the React Dev Tools work in Firefox.
+      # 'unsafe-eval' is needed because many libraries use some form of eval internally.
+      # The sha256 hash refers to the google analytics configuration script in main.scala.html.
+      script-src = "'self' 'unsafe-inline' 'unsafe-eval' *.googletagmanager.com app.olvy.co webknossos.olvy.co *.gstatic.com 'sha256-NA81o2FV8jh0UizSPsBz2qEGpXfap54Eghuo7vxrH8g='"
+      # 'unsafe-inline' is needed, because the olvy script uses inline CSS.
+      style-src = "'self' 'unsafe-inline' *.gstatic.com"
+      # In production ws://localhost:9002 should be removed as it is only needed to make the webpack dev-server work.
+      # api.github.com and gist.github.com are needed for the user script functionality.
+      # All datastore URLs need to be included here, for example: *.webknossos.org demo.data1-brain.esc.rzg.mpg.de
+      connect-src = "'self' ws://localhost:9002 app.olvy.co api.github.com gist.github.com *.google-analytics.com api.airbrake.io"
+      frame-src = "'self' webknossos.olvy.co"
+      # *.voxelytics.com is needed for images embedded in Voxelytics reports.
+      img-src = "'self' data: *.googletagmanager.com *.voxelytics.com"
+      media-src = "*.webknossos.org"
+      font-src = "'self' data:"
```

Comment:
LGTM

I assume, that these two directives were missing / causing errors in production?

Reply:
Yes, the media-src is for embedded videos from docs.webknossos.org and the font-src is needed for the dark theme (on Mac only for some reason).